shard.lua

```diff
@@ -331,23 +344,23 @@ local function shard_status()
         online = {},
         offline = {},
         uuids = {},
-        maintenance = maintenance
+        -- maintenance = maintenance
     }
     for _, shard_set in ipairs(shards) do
-         for _, shard in ipairs(shard_set) do
-             local s = { uri = shard.uri, id = shard.id }
-             if shard.conn and shard.conn:is_connected() then
-                 table.insert(result.online, s)
-                 local ok, info = pcall(shard.conn.call, shard.conn, 'box.info')
-                 if not ok then
-                     table.insert(result.offline, s)
-                 else
-                     result.uuids[info.uuid] = s.uri
-                 end
-             else
-                 table.insert(result.offline, s)
-             end
-         end
+        for _, shard in ipairs(shard_set) do
+            local s = { uri = shard.uri, id = shard.id }
+            if shard.conn and shard.state == 'connected' then
```

Comment:
I do not like the connpool and checks like this. It is possible, that a state is connected, but shard.conn is nil? If not, then why do you check it here? If yes, then what a sense in shard.state, if it is not synced with shard.conn?